sdk/detect/env.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package env provides a detector that loads resource information from
+// the OTEL_RESOURCE environment variable. A list of labels of the form
+// `<key1>=<value1>,<key2>=<value2>,...` is accepted. Domain names and
+// paths are accepted as label keys.
+package detect
```

Comment:
The docstring doesn't match the package declaration.  I think it's probably fine for this `Detector` to live in the `detect` package, but the documentation should match reality.